ntoskrnl/ex/time.c

```diff
@@ -468,6 +528,32 @@ NtSetSystemTime(IN PLARGE_INTEGER SystemTime,
         _SEH2_END;
     }
 
+    /* Read time zone information from the registry and set the clock */
+    Status = RtlQueryTimeZoneInformation(&TimeZoneInformation);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("RtlQueryTimeZoneInformation failed (Status 0x%08lx)\n", Status);
```

Comment:
As Serge pointed, here TimeZoneInformation is very certainly still non-initialized, so using it below is dangerous.
What do you want to do if querying the timezone info fails ?

Reply:
I intend to initialize this in its definition like the following:
```
    RTL_TIME_ZONE_INFORMATION TimeZoneInformation = { 0 };
```
Testing for zeroes (0) is done in some of this code to determine the code patch already.
Updates to follow shortly.

Reply:
Now done. Thanks.

Reply:
@Doug-Lyons I've just noticed that in the current PR this suggestion wasn't still applied; see my comments below.

Reply:
@HBelusca, Yes. I changed my mind and initialized it in the line numbers 471 and 472 above so that I could explicitly do it there and put a comment so that it would not be missed, but now I have restored it to what it was before because it is simpler this way. Thanks.